peer/brontide.go

```diff
@@ -2309,7 +2320,9 @@ func (p *Brontide) genDeliveryScript() ([]byte, error) {
 // channels maintained with the remote peer.
 //
 // NOTE: This method MUST be run as a goroutine.
-func (p *Brontide) channelManager() {
+func (p *Brontide) channelManager(client *subscribe.Client) {
+	// When exit, cancel the channel event subscription.
+	defer client.Cancel()
```

Comment:
Since the `reenableTimeout` only fires once, this `Cancel` should be called in that same block when the `reenableTimeout` is set to `nil`. If we call `Cancel` here, then this will cause updates to queue up in the internal overflow queue, possibly leading to DoS. This is because we may be done receiving channel events, but the p2p connection is still up and therefore the `channelManager` goroutine is still up.